src/sql/workbench/contrib/notebook/browser/outputs/gridOutput.component.ts

```diff
@@ -488,9 +488,19 @@ export class DataResourceDataProvider implements IGridDataProvider {
 		// interface than the query execution service's saveResults handlers. Here, we take the
 		// format-specific request params (eg, includeHeaders for CSV) and merge the format-agnostic
 		// request params for the serialization request (eg, saveFormat, filePath).
+		let providerId: string;
+		switch (this.cellModel.notebookModel.currentKernelAlias) {
+			case 'Kusto':
```

Comment:
This is pretty hacky and not friendly to new providers being added later. The NotebookModel already has a ton of stuff using providers - why not use those?

Reply:
These serialization providers have different IDs than the ones we use for notebook providers. For example, a SQL notebook uses the "MSSQL" serialization provider. We only use this service for notebook query results, so beyond Kusto and SQL it isn't used. Query Editor has its own RPC calls for exporting query results.

Reply:
Do we really not have a way for Notebook providers to be mapped to connection/serialization providers? That seems like something we'd need so we can properly route these serialization requests without having to update this hardcoded list every time a new provider was added...

Maybe this is something the Notebook provider can optionally provide - what serialization provider it should use (defaulting to MSSQL if none provided)

Reply:
Also Kusto and LogAnalytics are aliases of the SQL kernel, so they'd have the same provider

Reply:
Even if we have each provider specify a serializer, we still run into problems because Kusto and LogAnalytics are aliases for the SQL kernel and have the same provider

Reply:
Hrm, yeah definitely a pretty big mess. Well this at least improves the current experience so I guess it's fine to get in but if we end up having to come back here and special case more stuff we should definitely take some time to try and do this properly.

Reply:
Actually, it looks like these serializer IDs correspond to the connection providers, rather than the notebook ones. I'll use those instead.